src/lib/concurrency/transaction_manager.cpp

```diff
@@ -13,6 +12,7 @@ void TransactionManager::reset() {
   manager._next_transaction_id = INITIAL_TRANSACTION_ID;
   manager._last_commit_id = INITIAL_COMMIT_ID;
   manager._last_commit_context = std::make_shared<CommitContext>(INITIAL_COMMIT_ID);
+  manager._active_snapshot_commit_ids.clear();
```

Comment:
Just saw this. Does this actually make sense? Shouldn't we assert that this already is empty?

Reply:
True. This code was written before calling `deregister_transaction` from the `TransactionContext` destructor.
So I'll replace it as follows:
```suggestion
  DebugAssert(manager._active_snapshot_commit_ids.empty(),
          "Some transactions do not seem to have finished yet, as they are still registered as active.")
```